zoo/policies/open-agent/open_agent/agent.py

```diff
@@ -405,7 +405,8 @@ def update_impatience(self, ego_state):
 
     def act(self, obs):
         ego = obs.ego_vehicle_state
-        wps = obs.waypoint_paths[len(obs.waypoint_paths) // 2]
+        lane_index = min(len(obs.waypoint_paths) - 1, ego.lane_index)
+        wps = obs.waypoint_paths[lane_index]
```

Comment:
The intention of this change is to keep the OpEn stay in the current lane.

Reply:
@davidrusu this change needs a close look

Reply:
does lane index map directly to the waypoint path index? You may need to inspect the waypoints themselves to and look for a path that starts with a waypoint that has the correct edge id and lane index.

Reply:
I see. Need to check the lane_index of the first waypoint. Also need to ensure works at intersections.

Reply:
I loop through the `waypoint_paths` to see if I can match one, choose the first path that has the same lane_index.